redis/redisLimiter.go

```diff
@@ -195,6 +185,49 @@ func (rl *rateLimiter) getFailConfig(mode Mode) (time.Duration, int64) {
 	return rl.freezeFailureConfig.limitPeriod, rl.freezeFailureConfig.maxFailures
 }
 
+// SetSecurityModeNoExpire will set the key from volatile to persistent
+func (rl *rateLimiter) SetSecurityModeNoExpire(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	wasSet, err := rl.storer.SetPersist(ctx, key)
+	if err != nil {
+		return err
+	}
+
+	if !wasSet {
+		log.Debug("security mode was not set")
+	}
+
+	return nil
+}
+
+// UnsetSecurityModeNoExpire will set the key from persistent to volatile
+func (rl *rateLimiter) UnsetSecurityModeNoExpire(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	limitPeriod, _ := rl.getFailConfig(SecurityMode)
+
+	_, err := rl.storer.ExpireTime(ctx, key)
+	if err == nil {
+		return nil
+	}
+	if errors.Is(err, ErrKeyNotExists) {
+		return nil
+	}
```

Comment:
this lines should be removed.. SetExpireIfNotExists basically does this internally

Reply:
removed